sklearn/metrics/_ranking.py

```diff
@@ -1419,3 +1420,114 @@ def ndcg_score(y_true, y_score, k=None, sample_weight=None, ignore_ties=False):
+        Target scores. These can be either probability estimates or
+        non-thresholded decision values as returned by :term:`predict_proba` or
+        :term:`decision_function`, respectively.
+
+    k : int, default=2
+        Number of guesses allowed to find the correct label.
+
+    normalize : bool, default=True
+        If `True`, return the fraction of correctly classified samples.
+        Otherwise, return the number of correctly classified samples.
+
+    sample_weight : array-like of shape (n_samples,), default=None
+        Sample weights. If `None`, all samples are given the same weight.
+
+    Returns
+    -------
+    score : float
+        The top-k accuracy score. The best performance is 1 with
+        `normalize == True` and the number of samples with
+        `normalize == False`.
+
+    See also
+    --------
+    accuracy_score
+
+    Notes
+    -----
+    In cases where two or more labels are assigned equal predicted scores,
+    the labels with the smallest indices will be chosen first. This might
+    impact the result if the correct label falls after the threshold because
+    of that.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.metrics import top_k_accuracy_score
+    >>> y_true = np.array([0, 1, 2, 2])
+    >>> y_score = np.array([[0.5, 0.2, 0.2],  # 0 is in top 2
+    ...                     [0.3, 0.4, 0.2],  # 1 is in top 2
+    ...                     [0.2, 0.4, 0.3],  # 2 is in top 2
+    ...                     [0.7, 0.2, 0.1]]) # 2 isn't in top 2
+    >>> top_k_accuracy_score(y_true, y_score, k=2)
+    0.75
+
+    # Not normalizing gives the number of "correctly" classified samples
+    >>> top_k_accuracy_score(y_true, y_score, k=2, normalize=False)
+    3
+    >>> top_k_accuracy_score([0, 0, 0, 0], y_score, k=2)
+    0.75
+
+    """
+    if k == 1:
+        return accuracy_score(y_true, y_score, normalize, sample_weight)
+
+    check_consistent_length(y_true, y_score, sample_weight)
+    y_true_type = type_of_target(y_true)
+
+    if y_true_type not in {'binary', 'multiclass'}:
+        raise ValueError(
+            "'y_true' type must be 'binary' or 'multiclass', got "
+            f"'{y_true_type}' instead."
+        )
+
+    y_true = column_or_1d(y_true)
+    y_score = check_array(y_score)
+
+    n_classes = y_score.shape[1]
+    if k >= n_classes:
+        raise ValueError(
+            f"'k' ({k}) can't be greater than or equal to 'n_classes' "
+            f"({n_classes}). This will result in a perfect score and is "
+            "therefore meaningless."
+        )
+
+    n_classes_y_true = y_true.max() + 1
+    if n_classes_y_true > n_classes:
+        raise ValueError(
+            f"Number of classes in 'y_true' ({n_classes_y_true}) is greater "
+            f"than the number of columns in 'y_score' ({n_classes})."
+        )
+
+    sorted_pred = np.argsort(-y_score, axis=1)
+    hits = [y in pred[:k] for y, pred in zip(y_true, sorted_pred)]
```

Comment:
Ordinarily, we would use numpy operations to do this. I.e. `(y_true == pred[:, :k]).any(axis=1)`

Reply:
Doesn't work unfortunately. Couldn't find any numpy operation doing the job. `np.isin` is the closest I found, but it only allow the same set to compare each elements with... any idea?

Reply:
I remember seeing something like this somewhere, not sure where: `hits = (y_true == sorted_pred[:, :k].T).any(axis=0)`. Would this work?

Reply:
Works great! Thanks @averdones :+1:

Reply:
Cheers!